databox/indexer/src/handlers/phraseanet/indexer.ts

```diff
@@ -763,12 +763,29 @@ function translateDocumentSettings_toPdf(): object {
 function translateImageSettings(sd: PhraseanetSubdefStruct): object {
     // todo: extension ?
     const size = sd.options['size'];
+    let format: string;
+    switch (sd.options['icodec'] ?? '') {
+        case 'jpeg':
+            format = 'jpeg';
+            break;
+        case 'png':
+            format = 'png';
+            break;
+        case 'tiff':
+            format = 'tiff';
+            break;
+        default:
+            throw new Error(
+                `Unsupported image codec: ${sd.options['icodec']} for subdef image:${sd.name}`
+            );
+    }
```

Comment:
Prefer mapping over switch/case:

```suggestion
    if (!sd.options.icodec) {
        throw new Error(`Missing image codec for subdef image:${sd.name}`);
    }

    const icodecFormats = {
        jpeg: 'jpeg',
        png: 'png',
        tiff: 'tiff',
    }

    const format = icodecFormats[sd.options.icodec] ?? throw new Error(`Unsupported image codec: ${sd.options.icodec} for subdef image:${sd.name}`);
```

Reply:
`sd.options['icodec']` is always set (api response)
throw is no a rvalue...
```
Unexpected "throw"
const format = icodecFormats[sd.options.icodec] ?? throw new Error(`Unsupported image codec: ${sd.options.icodec} for subdef image:${sd.name}`);
```

Reply:
> sd.options['icodec'] is always set (api response)

OK, but you wrote `switch (sd.options['icodec'] ?? '')`

> Unexpected "throw"

Oops, it only works in PHP :(
So:

```ts
const format = icodecFormats[sd.options.icodec];
if (!format) {
    throw new Error(`Unsupported image codec: ${sd.options.icodec} for subdef image:${sd.name}`);
}
```